aptos-move/block-executor/src/view.rs

```diff
@@ -178,6 +186,7 @@ pub(crate) struct LatestView<'a, T: Transaction, S: TStateView<Key = T::Key>, X:
     base_view: &'a S,
     latest_view: ViewMapKind<'a, T, X>,
     txn_idx: TxnIndex,
+    id_counter: AtomicU32,
```

Comment:
LatestView and none of its members is not shared among threads, which can also be seen because the type and none of the members are Sync. Even MVHashMapView isn't shared / Sync, MVHashMap reference that it holds is the only shared data-structure (reference) in this file.

We wouldn't want to implement a shared counter in MVHashMap, but we can follow the same blueprint - create in executor, and pass a reference (or an Arc, but shouldn't need Arc) to the constructor of LatestView - i.e. right now every transaction is creating a new AtomicU32 in the new_mv_view and new_btree_view calls, instead, we can pass & there.

Ideally, the counter used for sequential execution would be a different type, which can be accomplished by putting & to the counter not inside LatestView, but Inside the variants of the ViewMapKind. MultiVersion would have &AtomicU32 and Unsync would have &u32.